### PR TITLE
KFSPTS-29283 Fix missing Doc Reference Info for ISO 20022

### PR DESCRIPTION
**Pull request by cah292**
This PR fixes the ISO 20022 files so that the Invoice/Credit Note indicator will get filled in on payments sourced from PDP feeds. It also allows for placing the KFS doc number into the invoice number field when necessary, similar to (but not exactly like) the functionality in our FastTrack file generation code.

### CODE REVIEW

File: src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java [hunk 1]
```diff
@@ -1527,11 +1530,12 @@ private static DocumentType5Code determineDocumentTypeCode(
...
         } else {
-            LOG.warn(
-                    "determineDocumentTypeCode(...) - PaymentDetail has unexpected DocumentTypeCode : "
-                    + "paymentDetailFinancialDocumentTypeCode={}",
-                    paymentDetail::getFinancialDocumentTypeCode
-            );
+            /*
+             * CU Customization: Allow payments from PDP feeds to be marked as Invoices or Credit Notes.
+             */
+            documentTypeCode = paymentDetail.getNetPaymentAmount().isGreaterEqual(KualiDecimal.ZERO)
```
Comment (db483) [inline src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java:1536]: Is the FinancialDocumentTypeCode value not filled when the payment comes from the PDP feed? I was just wondering if there is a way to determine that the payment comes from the PDP feed or is enough to assume that the payment comes from the PDP feed if it is not invoice or credit note? Thank you

File: src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java [hunk 2]
```diff
@@ -1557,6 +1561,26 @@ private static boolean isCreditNote(
...
+    private static String determineReferredDocumentNumber(
+            final PaymentDetail paymentDetail
+    ) {
+        if (StringUtils.equalsAny(paymentDetail.getFinancialDocumentTypeCode(),
+                DisbursementVoucherConstants.DOCUMENT_TYPE_CHECKACH,
+                DisbursementVoucherConstants.DOCUMENT_TYPE_CODE)) {
+            if (StringUtils.isNotBlank(paymentDetail.getInvoiceNbr()) && paymentDetail.getInvoiceDate() != null) {
+                return paymentDetail.getInvoiceNbr();
+            } else {
+                return CuDisbursementVoucherConstants.DV_EXTRACT_EDOC_NUMBER_PREFIX_IDENTIFIER
```
Comment (db483) [inline src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java:1576]: what is the source of this payment?

**Review by db483 — CHANGES_REQUESTED** [review]
Changes look good but I had a couple of questions. Thank you.

File: src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java (hunk 2 above)
Comment (cah292) [reply src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java:1576]: As per the `StringUtils.isAny()` comparison a few lines above, the payment is sourced from a DV if execution reaches this marked line. As far as I can tell, DV-sourced payments appear to be the only ones that can potentially have a blank invoice number.

Please let me know if you need me to make the is-DV check clearer.

File: src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java (hunk 1 above)
Comment (db483) [reply src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java:1536]: Given that the FastTrack file generation worked in a similar way I am ok with this assumption. Thank you.

File: src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java (hunk 2 above)
Comment (db483) [reply src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java:1576]: Are these DV's created in a different manner? Or is just a condition that can appear on any DV? Thank you.

Comment (cah292) [reply src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java:1576]: DVs have fields in the "Payment Information" section for specifying the Invoice Number and Invoice Date, but they appear to be optional fields. As a result, there are cases where a DV document does not have an Invoice Number.

**Review by db483 — APPROVED** [review]
Changes look good and unit tests pass.

**Pull request merged**